docs/environment-variables.md

```diff
@@ -0,0 +1,35 @@
+# Environment Variables
+
+This document outlines the set of environment variables that can be used to customize the behaviours at different levels.
+These environment variables are typically added to test out experimental features and should not be needed by most users.
+
+## Controller
+
+| Name | Type | Description|
+|----------|------|------------|
+| `ALL_POD_CHANGES_SIGNIFICANT` | bool |  Whether to consider all pod changes as significant during pod reconciliation. |
```

Comment:
I would also wrap types in in-line codeblocks to be consistent with our docs elsewhere

```suggestion
| `ALL_POD_CHANGES_SIGNIFICANT` | `bool` |  Whether to consider all pod changes as significant during pod reconciliation. |
```